guide/src/pagination.md

```diff
@@ -0,0 +1,127 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+# Working with List operations
+
+Some services return over potentially large lists of items, such as rows or
+resource descriptions. To keep CPU and memory usage under control, the services
+returned theses resources in `pages`: smaller subsets of the items with a
+continuation token to request the next subset.
+
+Iterating over items in this way can be tedious. The client libraries provide
+adapters to convert the pages into asynchronous iterators. This guide will show
+you how to work with these adapters.
+
+## Prerequisites
+
+The guide uses the [Secret Manager] service. That makes the examples more
+concrete and therefore easier to follow. With that said, the same ideas work for
+any other service.
+
+You may want to follow the service [quickstart]. This guide will walk you
+through the steps necessary to enable the service, ensure you have logged in,
+and that your account has the necessary permissions.
+
+## Dependencies
+
+As it is usual with Rust, you must declare the dependency in your
+`Cargo.toml` file. We use:
+
+```toml
+{{#include ../samples/Cargo.toml:secretmanager}}
+```
+
+## Iterating List methods
+
+To help iterate the items in a list method, the API return an implementation of
+the `Paginator` trait. We need to introduce it in scope via a `use` declaration:
+
+```rust,ignore
+{{#include ../samples/src/pagination.rs:paginator-use}}
+```
+
+To iterate the items, we use the provided Paginator `items` function.
+
+```rust,ignore
+{{#include ../samples/src/pagination.rs:paginator-iterate-items}}
+```
+
+In rare cases, pages contain extra information outside of the items that you may
+need access to. Or you may need to checkpoint your progress across processes. In
+these cases, you can do so by iterating over the full pages instead of the
+individual items.
+
+```rust,ignore
+{{#include ../samples/src/pagination.rs:paginator-iterate-pages}}
+```
+
+### Working with futures::Stream
+
+You may want to use these APIs in the larger Rust ecosystem of asynchronous
+streams, such as `tokio::Stream`.  This is readily done, but you must first
+enable the `unstable-streams` feature in the `google_cloud_gax` crate:
+
+```toml
+{{#include ../samples/Cargo.toml:gax-with-streams}}
+```
+
+The name of this feature is intended to convey that we consider these APIs
+unstable, because they are! You should only use them if you are prepared to work
+with deal with any breaks that result from incompatible changes to the
+[`futures::Stream`][future-stub] trait.
+
+The examples will also use the `futures::stream::StreamExt` trait, so we must
+add the crate that defines it.
+
+```toml
+{{#include ../samples/Cargo.toml:futures}}
+```
+
+```rust,ignore
+{{#include ../samples/src/pagination.rs:paginator-stream-items}}
+```
+
+Similarly, use the `into_stream` function to stream the pages.
```

Comment:
nit: If you take the above suggestion, then maybe:

"Similarly, we can use the ..."